tpp-app/src/home/pages/LogSymptomsScreen.js

```diff
@@ -1,18 +1,341 @@
+    Alert.alert(
+      info.title,
+      info.message,
+      [
+        {
+          text: info.cancelTitle,
+          onPress: () => reject(),
+          style: "cancel"
+        },
+        { text: info.acceptTitle, onPress: () => resolve() }
+      ]
+    );
+  })
+
+  // Helper function to set form state back to default
+  const resetForm = async (newDate) => {
+    changeDate(newDate);
+    setFetching(true); // triggers useEffect to fetch symptoms of newDate
+    setDirty(false);
+    setSubmitting(false);
+  }
+
+  // Change the selected date to log symptoms for
+  const switchDate = async (goFwd) => {
+    let newDate = selectedDate;
+    newDate.setDate(goFwd ? newDate.getDate() + 1 : newDate.getDate() - 1);
+
+    if (isDirty) {
+      alertPopup(unsavedChanges)
+        .then(async () => { // YES switch date
+          await resetForm(newDate);
+        })
+        .catch() // CANCEL do nothing and close alert
+    } else {
+      await resetForm(newDate);
+    }
+  }
 
-  // this is how you access the params passed from the previous page
-  // const date = route.params.date
-  // console.log(date) 
-  // {year: 2022, month: 2, day: 22, timestamp: 1645488000000, dateString: "2022-02-22"}
 
   return (
-    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center', borderRadius: '20px 20px 0px 0px' }}>
-      <Text style={{ fontSize: 20, textAlign: 'center' }}>Log your symptoms here. Swipe down to dismiss or click the button.</Text>
-      <Button onPress={() => navigation.goBack()} title="Dismiss" />
-    </View>
+    <SafeAreaView style={styles.screen}><ScrollView>
+
+      {/* HEADER NAV */}
+      <View style={styles.navbarContainer}>
+
+          {/* CLOSE BUTTON */}
+          <TouchableOpacity
+            onPress={() => {
+              if (isDirty) {
+                alertPopup(unsavedChanges)
+                  .then(() => { // YES discard changes
+                    navigation.goBack();
+                  })
+                  .catch() // CANCEL do nothing and close alert
+              } else {
+                navigation.goBack();
+              }
+            }}
+            style={styles.close}>
+              <CloseIcon fill={'#000000'}/>
+          </TouchableOpacity>
+
+          {/* SWITCH AND DISPLAY DATE */}
+          <View style={styles.switchDate}>
+            <DateArrow
+              onPress={async () => await switchDate(false)}
+              isRight={false}
+            />
+            <View style={styles.centerText}>
+              <Text style={styles.subtitle}>Log your symptoms for:</Text>
+              <Text style={styles.navbarTitle}>{getDateString(selectedDate, 'MM DD, YYYY')}</Text>
+            </View>
+            <DateArrow
+              onPress={async () => await switchDate(true)}
+              isRight={true}
+            />
+          </View>
+
+      </View>
+
+      {/* SYMPTOM ACCORDIONS */}
+      {symptoms.map((symptom, i) => {
+        if (trackingPrefs[i])
```

Comment:
See my previous comment, you will probably be wanting to check that trackingPrefs[i][1] is true 

checking trackingPrefs[i] also relies on what's being returned from GETAllTrackingPreferences and symptoms being in the same order. For now it's fine, but I think if you matched the key and then looked at the value it would be safer.

Reply:
Changed now!